### PR TITLE
Bm25 indexing

### PR DESCRIPTION
**Pull request by bhavani105**

### CODE REVIEW

**Comment by bhavani105** [comment]
Add building index using pyserini
Integrate BM25 index and search into run_ir.py

File: examples/ir/run_ir.py [hunk 1]
```diff
@@ -140,7 +136,18 @@ def main():
     elif process_args.engine_type == 'DPR':
         pass
     elif process_args.engine_type == 'BM25':
-        pass
+        logger.info(f"Running BM25")
+
+        engine = BM25Engine(bm25_args)
+        
+        if hasattr(process_args, 'do_index') and process_args.do_index:
```
Comment (franzmpub) [inline examples/ir/run_ir.py:143]:
```suggestion
        if hasattr(process_args, 'do_index') and process_args.do_index:
            logger.info("Running BM25 search")
```

File: examples/ir/run_ir.py [hunk 2]
```diff
@@ -37,25 +41,17 @@ class DPRConfig:
...
         # let's parse it to get our arguments.
         process_args, model_args, training_args = \
             parser.parse_json_file(json_file=os.path.abspath(sys.argv[1]))
     else:
         #process_args, colbert_args, dpr_args, bm25_args = parser.parse_args_into_dataclasses()
-        (process_args, remaining_args) = parser.parse_args_into_dataclasses(return_remaining_strings=True)
+        (process_args, bm25_args, remaining_args) = parser.parse_args_into_dataclasses(return_remaining_strings=True)
 
     if process_args.engine_type == 'ColBERT':
         from primeqa.ir.dense.colbert_top.colbert.infra import Run, RunConfig
```
Comment (franzmpub) [inline examples/ir/run_ir.py:57]:
```suggestion
        logger.info(f"Running ColBERT")
        from primeqa.ir.dense.colbert_top.colbert.infra import Run, RunConfig
```

File: examples/ir/run_ir.py [hunk 3]
```diff
@@ -140,7 +136,18 @@ def main():
     elif process_args.engine_type == 'DPR':
         pass
     elif process_args.engine_type == 'BM25':
-        pass
+        logger.info(f"Running BM25")
+
+        engine = BM25Engine(bm25_args)
+        
+        if hasattr(process_args, 'do_index') and process_args.do_index:
+            engine.do_index()
```
Comment (franzmpub) [inline examples/ir/run_ir.py:144]:
```suggestion
            engine.do_index()
             logger.info(f"BM25 indexing finished")
```

**Review by franzmpub — APPROVED** [review]
LGTM, left a few suggestions related to logging

**Review by yulongl-ibm — APPROVED** [review]

**Review by mihaela-bornea — APPROVED** [review]

**Pull request merged**